### PR TITLE
Update dotnet.yaml

### PR DESCRIPTION
**Pull request by kasperk81**
release: https://devblogs.microsoft.com/dotnet/announcing-dotnet-7/ (6.0 is still the LTS)

### CODE REVIEW

**Comment by RubenRBS** [comment]
Hi! Thanks for taking the time to keep dotnet updated. 

We usually want to keep old targets laying around, as we don't know the differences between versions

File: bin/yaml/dotnet.yaml [hunk 1]
```diff
@@ -8,7 +8,8 @@ compilers:
       - -c
       - "/usr/bin/env DOTNET_CLI_HOME=/tmp /opt/compiler-explorer/{dir}/bin/coreclr/Linux.x64.Checked/.dotnet/dotnet --version"
     targets:
-      - v6.0.1
+      - v6.0.11
```
Comment (kasperk81) [inline bin/yaml/dotnet.yaml:11]:
```suggestion
      - v6.0.1
      - v6.0.11
```

File: bin/yaml/dotnet.yaml [hunk 2]
```diff
@@ -18,4 +19,5 @@ compilers:
       - -c
       - "cat /opt/compiler-explorer/{dir}/packages/microsoft.netcore.app.runtime.linux-x64/6.0.0/Microsoft.NETCore.App.versions.txt"
     targets:
-      - v6.0.0
+      - v6.0.11
```
Comment (kasperk81) [inline bin/yaml/dotnet.yaml:22]:
```suggestion
      - v6.0.0
      - v6.0.11
```

**Comment by kasperk81** [comment]
@RubenRBS, i've added the old version.
but in .net, one version band (say `6.x`) only need the latest version as all version updates are 'servicing fixes' on after the rtm.

**Comment by kasperk81** [comment]
Q. can we control which version will be the default in ui? i think people would prefer the latest (7.0.0) to be selected for dotnet languages.

**Comment by RubenRBS** [comment]
Yes! But that's something that happens in the main repo. As this is the first time dotnet has been updated, I'll try to get it updated and link the corresponding PR here so for the next update there's a template to follow

**Comment by kasperk81** [comment]
@RubenRBS, any updates?

**Review by mattgodbolt — APPROVED** [review]
We're going to give this a go - we may have some changes but if they're simple we can make them! thanks for your patience

**Pull request merged**